bundles/org.openhab.binding.juicenet/src/main/java/org/openhab/binding/juicenet/internal/handler/JuiceNetBridgeHandler.java

```diff
@@ -0,0 +1,158 @@
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link JuiceNetBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Jeff James - Initial contribution
+ */
+@NonNullByDefault
+public class JuiceNetBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(JuiceNetBridgeHandler.class);
+
+    protected JuiceNetBridgeConfiguration config = new JuiceNetBridgeConfiguration();
+    private final JuiceNetApi api = new JuiceNetApi();
+
+    public JuiceNetApi getApi() {
+        return api;
+    }
+
+    protected @Nullable ScheduledFuture<?> pollingJob;
+    protected List<JuiceNetApi.JuiceNetApiDevice> listDevices = Collections.<JuiceNetApi.JuiceNetApiDevice> emptyList();
+    protected @Nullable JuiceNetDiscoveryService discoveryService;
+
+    public JuiceNetBridgeHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(JuiceNetBridgeConfiguration.class);
+
+        logger.trace("JuiceNetBridgeHandler:initialize");
+
+        try {
+            api.initialize(config.api_token, this.getThing().getUID());
+        } catch (JuiceNetApiException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR, e.toString());
+            return;
+        }
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        scheduler.execute(() -> {
+            try {
+                listDevices = api.queryDeviceList();
+            } catch (JuiceNetApiException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR, e.toString());
+                return;
+            } catch (IOException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR, e.toString());
+                return;
+            } catch (InterruptedException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR, e.toString());
+                return;
+            }
+
+            for (JuiceNetApi.JuiceNetApiDevice dev : listDevices) {
+                discoveryService.notifyDiscoveryDevice(dev.unit_id, dev.name, dev.token);
+            }
+
+            pollingJob = scheduler.scheduleWithFixedDelay(this::pollDevices, 60, config.refreshInterval,
+                    TimeUnit.SECONDS);
+
+            logger.trace("Bridge is ONLINE");
```

Comment:
This logging can be omitted since the status change will be logged by the framework.